ops/model.py

```diff
@@ -2950,8 +2949,9 @@ def _format_action_result_dict(input: Dict[str, Any],
             # other exceptions raised on key validation...
             raise ValueError(f'invalid key {key!r}; must be a string')
         if not _ACTION_RESULT_KEY_REGEX.match(key):
-            raise ValueError("key '{!r}' is invalid: must be similar to 'key', 'some-key2', or "
-                             "'some.key'".format(key))
+            raise ValueError(
+                f"key '{key!r}' is invalid: must be similar to 'key', 'some-key2', "
+                "or 'some.key'")
```

Comment:
And more :) - this oddly has repr() but also explicit `'` - I think that's a mistake that we can fix here.

```suggestion
                f"key {key!r} is invalid: must be similar to 'key', 'some-key2', "
                f"or 'some.key'")
```